omnisend/includes/Internal/V1/class-client.php

```diff
@@ -94,6 +93,126 @@ public function create_contact( $contact ): CreateContactResponse {
+		$response = wp_remote_post(
+			OMNISEND_CORE_API_V5 . '/contacts',
+			array(
+				'body'    => wp_json_encode( $contract_array ),
+				'headers' => array(
+					'Content-Type'          => 'application/json',
+					'X-API-Key'             => $this->api_key,
+					'X-INTEGRATION-NAME'    => $this->plugin_name,
+					'X-INTEGRATION-VERSION' => $this->plugin_version,
+				),
+				'timeout' => 10,
+			)
+		);
+
+		if ( is_wp_error( $response ) ) {
+			error_log( 'wp_remote_post error: ' . $response->get_error_message() );
+			return new SaveContactResponse( '', $response );
+		}
+
+		$http_code = wp_remote_retrieve_response_code( $response );
+
+		if ( $http_code >= 400 ) {
+			$body    = wp_remote_retrieve_body( $response );
+			$err_msg = "HTTP error: {$http_code} - " . wp_remote_retrieve_response_message( $response ) . " - {$body}";
+			$error->add( 'omnisend_api', $err_msg );
+			return new SaveContactResponse( '', $error );
+		}
+
+		$body = wp_remote_retrieve_body( $response );
+		if ( ! $body ) {
+			$error->add( 'omnisend_api', 'empty response' );
+			return new SaveContactResponse( '', $error );
+		}
+
+		$arr = json_decode( $body, true );
+
+		if ( empty( $arr['contactID'] ) ) {
+			$error->add( 'omnisend_api', 'contactID not found in response.' );
+			return new SaveContactResponse( '', $error );
+		}
+
+		return new SaveContactResponse( (string) $arr['contactID'], $error );
+	}
+
+	public function get_contact_by_email( string $email ): GetContactResponse {
+		$error = new WP_Error();
+
+		$response = wp_remote_get(
+			OMNISEND_CORE_API_V5 . '/contacts?email=' . $email,
+			array(
+				'headers' => array(
+					'Content-Type'          => 'application/json',
+					'X-API-Key'             => $this->api_key,
+					'X-INTEGRATION-NAME'    => $this->plugin_name,
+					'X-INTEGRATION-VERSION' => $this->plugin_version,
+				),
+				'timeout' => 10,
+			)
+		);
+
+		if ( is_wp_error( $response ) ) {
+            error_log('wp_remote_post error: ' . $response->get_error_message());
+
+			return new GetContactResponse( null, $error );
+		}
+
+		$http_code = wp_remote_retrieve_response_code( $response );
+		if ( $http_code >= 400 ) {
+			$body    = wp_remote_retrieve_body( $response );
+			$err_msg = "HTTP error: {$http_code} - " . wp_remote_retrieve_response_message( $response ) . " - {$body}";
+			$error->add( 'omnisend_api', $err_msg );
+
+			return new GetContactResponse( null, $error );
+		}
+
+		$body = wp_remote_retrieve_body( $response );
+		if ( ! $body ) {
+			$error->add( 'omnisend_api', 'empty response' );
+
+			return new GetContactResponse( null, $error );
+		}
+
+		$contact_data = json_decode( $body, true );
+
+		if ( empty( $contact_data['contacts'][0]['contactID'] ) ) {
+			$error->add( 'omnisend_api', 'contactID not found in response.' );
+
+			return new GetContactResponse( null, $error );
+		}
+
+		$contact = ContactFactory::create_contact( $contact_data['contacts'][0] );
+
+		if ( ! isset( $contact_data['contacts'][0] ) ) {
```

Comment:
Why this if is needed? I guess this will never be true, as previous check few lines above will do the work?
```
		if ( empty( $contact_data['contacts'][0]['contactID'] ) ) {
			$error->add( 'omnisend_api', 'contactID not found in response.' );

			return new GetContactResponse( null, $error );
		}
```

Reply:
correct this might be overdone with check. Removed